util/logtools/splitter.ts

```diff
@@ -317,9 +317,20 @@ export default class Splitter {
     return lines;
   }
 
+  processWithAnalysisOnly(line: string): string {
+    const splitLine = line.split('|');
+    const typeField = splitLine[0];
+    const filteredLine = this.doAnalysisFilter ? this.analysisFilter(line, typeField) : line;
+    return filteredLine ?? line;
```

Comment:
This one makes sense but it took a little mulling it over. It's out of scope for this PR, but I wonder whether it would be better to modify `analysisFilter()` to return a bool instead. Anything that consumes its output currently will necessarily have to do some additional bookkeeping to deal with `undefined`, so we might as well make that bookkeeping explicit and focused on "does the line go through or not".

Reply:
Yeah, I think that

> This one makes sense but it took a little mulling it over. It's out of scope for this PR, but I wonder whether it would be better to modify `analysisFilter()` to return a bool instead. Anything that consumes its output currently will necessarily have to do some additional bookkeeping to deal with `undefined`, so we might as well make that bookkeeping explicit and focused on "does the line go through or not".

Yeah, I think that's much cleaner, especially since the result is binary: either output the line or don't.  Linking this to #94 since that'll (probably) be the next thing I work on; I'll include this suggestion in that work, since it's going to involve refactoring a good chunk of the splitter code anyway.